torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -558,13 +558,18 @@ void enableProfiler(
 
   TORCH_CHECK(
       config.state == ProfilerState::KINETO ||
-      config.state == ProfilerState::KINETO_GPU_FALLBACK ||
-      config.state == ProfilerState::KINETO_ONDEMAND);
+      config.state == ProfilerState::KINETO_GPU_FALLBACK || config.global());
   TORCH_CHECK(
       !activities.empty(), "No activities specified for Kineto profiler");
 
-  if (config.state == ProfilerState::KINETO ||
-      config.state == ProfilerState::KINETO_GPU_FALLBACK) {
+  if (config.global()) {
```

Comment:
Should this line be like below? `ProfilerState::KINETO` seems to be missed.
```
if (config.state == ProfilerState::KINETO || config.global()) {
```

Reply:
No. `config.global()` is on-demand. `ProfilerState::KINETO` and `ProfilerState::KINETO_GPU_FALLBACK` are covered by the else branch.